Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,34 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+# True if this version of Ruby supports text encodings in its strings
+def ruby_has_encoding?
+  String.method_defined?(:force_encoding)
+end
+
+# Truncates a text string to fit within a byte size constraint,
+# preserving character encoding validity. The returned string will
+# be not much longer than the specified max_bytes, though the exact
+# shortfall or overrun may vary.
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options.fetch(:front_weight, 0.5)
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0.0 || front_weight > 1.0
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  glue = "\n[...snip...]\n"
+  max_bytes_in = [max_bytes - glue.bytesize, 1].max
+  bytes = ruby_has_encoding? ? s.dup.force_encoding("BINARY") : s
+  glue_bytes = ruby_has_encoding? ? glue.encode("BINARY") : glue
```

Comment:
Given these both use ternary operators with the same conditional: an `if/else` would be nice.